Libraries/FBReactNativeSpec/FBReactNativeSpec.podspec

```diff
@@ -44,4 +51,13 @@ Pod::Spec.new do |s|
   s.dependency "React-Core", version
   s.dependency "React-jsi", version
   s.dependency "ReactCommon/turbomodule/core", version
+
+  s.prepare_command = codegenCommand
+  s.script_phase = {
+    :name => 'Generate Native Modules Code',
+    :input_files => [File.join(__dir__, "..")],
+    :output_files => [File.join(__dir__, "Libraries", "FBReactNativeSpec", "FBReactNativeSpec", "FBReactNativeSpec.h"), File.join(__dir__, "Libraries", "FBReactNativeSpec", "FBReactNativeSpec", "FBReactNativeSpec-generated.mm")],
+    :script => codegenCommand,
+    :execution_position => :before_compile
+  }
```

Comment:
Technically, the codegen does not need to run during `pod install` or `pod update`, because it will be invoked by Xcode as needed through the injected build phase (see `s.script_phase`). However, if the generated files do not exist when the FBReactNativeSpec Pod is built, then the sources glob in the Podspec will not match any files and the resulting Pods project will lack any references to FBReactNativeSpec.h or FBReactNativeSpec-generated.mm.

By using prepare_command, we can ensure these files exist because the codegen will run before the FBReactNativeSpec Pod is built.

We then use input_files and output_files to provide hints to Xcode as to whether running the codegen again at build time is necessary: if the output files are not present, the codegen must run; if the input files have changed, the codegen must run; if neither condition is met, the codegen can be skipped.

Reply:
This is correct, CocoaPods will only include files that exist on disk at the time of running `pod install`.

Your solution is good, but I wonder if it would be helpful to reduce the number of places from where codegen is invoked, i.e. to only run it as part of the Xcode build? We should be able to make this work by changing the prepare command to `touch` those paths so they _exist_ but are simply empty files. What do you think?

Reply:
`touch` works for me as well. To ensure the script phase runs, I'll remove the codegen files from output_files, otherwise Xcode might skip the build phase.